nova/objects/flavor.py

```diff
@@ -612,6 +614,22 @@ def _flavor_get_all_from_db(context, inactive, filters, sort_key, sort_dir,
             query = query.filter(or_(*the_filter))
         else:
             query = query.filter(the_filter[0])
+
+    if 'hidden' in filters and filters['hidden'] is not None:
+        lowercase_extraspec_value_column = \
+            func.lower(api_models.FlavorExtraSpecs.value)
+        if filters['hidden'] is False:
```

Comment:
Why not go with this?
```python
if filters['hidden']:
  …
else:
  …
```

Reply:
I wanted to have the regular case (hidden = false) on top, and the unusual case (showing _only_ hidden flavors) last. The latter is also not used anywhere in the code currently.

I went with
```python
if not filters['hidden']
    ...
else:
    ...
```